File: DesignPatternsDotNet/BuilderPattern/MobileBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace BuilderPattern 
{
    class MobileBuilder : IMobileBuilder
    {
        private Mobile aMobile=null;
	    public MobileBuilder() {
	       aMobile=new Mobile();
	    }
	public void BuildKeypad(){
		aMobile.Keypad="QWERTY-Pad";
	}
	public void BuildSpeaker(){
		aMobile.Speaker="Loude Speakers ";
	}	
	public void BuildDisplayUnit(){
		aMobile.Display="Touch Display";
	}
	public void BuildMic(){
		aMobile.Mike="High sensored mike";
	}
	public void BuildNetwrkRadio(){
		aMobile.Network="3g Network";
	}
	public Mobile Mobile {
        get { return aMobile; }
	}
}
    
}

File: DesignPatternsJava/src/main/java/com/builder/MobileBuilder.java
package com.builder;

public class MobileBuilder implements IMobileBuilder{
	private Mobile aMobile=null;
	public MobileBuilder() {
	aMobile=new Mobile();
	}
	public void buildKeypad(){
		aMobile.setKeypad("QWERTY-Pad");
	}
	public void buildSpeaker(){
		aMobile.setSpeaker("Loude Speakers ");
	}	
	public void buildDisplayUnit(){
		aMobile.setDisplayUnit("Touch Display");
	}
	public void buildMic(){
		aMobile.setMic("High sensored mike");
	}
	public void buildNetwrkRadio(){
		aMobile.setNetwrkRadio("3g Network");
	}
	public Mobile getMobile() {
		return aMobile;
	}
}
